modules/middleware/src/buildQuery/index.js

```diff
@@ -190,7 +190,7 @@ function getGroupFilter({ nestedFields, filter: { content, op, pivot } }) {
   const applyBooleanWrapper = wrappers[op];
   const esFilters = content.map((filter) => opSwitch({ nestedFields, filter }));
   const isNested = !!esFilters[0]?.nested;
-  if (isNested && pivot === esFilters[0]?.nested.path) {
+  if (isNested && esFilters.map((f) => f.nested?.path).includes(pivot)) {
```

Comment:
This definitely needed changing to remove the impact of ordering of the entries in the filter. What I'm trying to determine is if there is any reason to check that the filters include that path.. if the sqon has the `pivot` property then we should use it without any other checks. So, why not just:
`if (isNested && pivot) {`